os/services/rpl-border-router/native/tun-bridge.c

```diff
@@ -33,166 +33,71 @@
  *         Joakim Eriksson <joakime@sics.se>
  */
 
-#include "net/ipv6/uip.h"
-#include "net/ipv6/uip-ds6.h"
 #include <stdio.h>
-#include <stdlib.h>
-#include <stdarg.h>
-#include <string.h>
 #include <sys/time.h>
-#include <sys/types.h>
-
-#include <unistd.h>
-#include <errno.h>
-#include <fcntl.h>
-#include <signal.h>
-#include <termios.h>
-#include <net/if.h>
-#include <sys/ioctl.h>
-#include <sys/socket.h>
+#include <err.h>
+#include "border-router.h"
+#include "tun6-net.h"
 
 /*---------------------------------------------------------------------------*/
 /* Log configuration */
 #include "sys/log.h"
 #define LOG_MODULE "BR"
 #define LOG_LEVEL LOG_LEVEL_NONE
 
-#ifdef linux
-#include <linux/if.h>
-#include <linux/if_tun.h>
-#endif
-
-#include <err.h>
-#include "net/netstack.h"
-#include "net/packetbuf.h"
-#include "cmd.h"
-#include "border-router.h"
-
-extern const char *slip_config_ipaddr;
-extern char slip_config_tundev[IFNAMSIZ + 1];
 extern uint16_t slip_config_basedelay;
-
-static int tunfd;
-
-static int set_fd(fd_set *rset, fd_set *wset);
-static void handle_fd(fd_set *rset, fd_set *wset);
-static const struct select_callback tun_select_callback = {
-  set_fd,
-  handle_fd
-};
-
-int ssystem(const char *fmt, ...)
-     __attribute__((__format__ (__printf__, 1, 2)));
-void ifconf_cleanup(const char *dev);
-void ifconf(const char *tundev, const char *ipaddr);
-int devopen(const char *dev, int flags);
-
-
-/*---------------------------------------------------------------------------*/
-void
-cleanup(void)
-{
-  ifconf_cleanup(slip_config_tundev);
-}
 /*---------------------------------------------------------------------------*/
-void CC_NORETURN
-sigcleanup(int signo)
-{
-  fprintf(stderr, "signal %d\n", signo);
-  exit(0);			/* exit(0) will call cleanup() */
-}
+static uint16_t delaymsec = 0;
+static uint32_t delaystartsec, delaystartmsec;
 /*---------------------------------------------------------------------------*/
-#ifdef linux
-int
-tun_alloc(char *dev, uint16_t devsize)
+static void
+tun_input_callback(void)
 {
-  struct ifreq ifr;
-  int fd, err;
-  LOG_INFO("Opening: %s\n", dev);
-  if((fd = open("/dev/net/tun", O_RDWR)) < 0) {
-    /* Error message handled by caller */
-    return -1;
+  /* Optional delay between outgoing packets */
+  /* Base delay times number of 6lowpan fragments to be sent */
+  /* delaymsec = 10; */
+  if(delaymsec) {
+    struct timeval tv;
+    int dmsec;
+    gettimeofday(&tv, NULL);
+    dmsec = (tv.tv_sec - delaystartsec) * 1000 + tv.tv_usec / 1000 - delaystartmsec;
```

Comment:
Can `int` be squeezed in at the start of the line without forcing line-wrapping?

Reply:
No, the line was already a little too long. I broke it into two lines.